app/views/shipit/tasks/new.html.erb

```diff
@@ -13,13 +13,32 @@
 
   <%= render_checklist @task.checklist %>
 
-  <section class="submit-section">
-    <%= form_for @task, url: stack_tasks_path(@stack, definition_id: @task.definition.id) do |f| %>
+  <%= form_for @task, url: stack_tasks_path(@stack, definition_id: @task.definition.id) do |f| %>
+
+    <% unless @task.definition.variables.empty? %>
+      <section>
+        </br>
+        <header class="section-header environment-variables">
+          <h2>Environment variables</h2>
+        </header>
+        <%= f.fields_for :env do |env_fields| %>
+          <% @task.definition.variables.each do |variable| %>
+            <p class="environment-variable">
+              <%= env_fields.text_field variable.name, value: variable.default %>
+              <%= env_fields.label variable.name, variable.title || variable.name %>
+            </p>
+          <% end %>
+        <% end %>
+      </section>
+    <% end %>
```

Comment:
I could. I didn't because I couldn't see what else could use it in the future. But I guess a form renderer for an array of `VariableDefinition`s could be handy